teos/appointments_dbm.py

```diff
@@ -0,0 +1,508 @@
+
+        Args:
+            prefix (:obj:`str`): the prefix of the data to load.
+
+        Returns:
+            :obj:`dict`: A dictionary containing the requested data (appointments or trackers) indexed by ``uuid``.
+
+            Returns an empty dictionary if no data is found.
+        """
+
+        data = {}
+
+        for k, v in self.db.iterator(prefix=prefix.encode("utf-8")):
+            # Get uuid and appointment_data from the db
+            uuid = k[len(prefix) :].decode("utf-8")
+            data[uuid] = json.loads(v)
+
+        return data
+
+    def get_last_known_block(self, key):
+        """
+        Loads the last known block given a key.
+
+        Args:
+            key (:obj:`str`): the identifier of the db to look into (either ``WATCHER_LAST_BLOCK_KEY`` or
+            ``RESPONDER_LAST_BLOCK_KEY``).
+
+        Returns:
+            :obj:`str` or :obj:`None`: A 16-byte hex-encoded str representing the last known block hash.
+
+            Returns ``None`` if the entry is not found.
+        """
+
+        last_block = self.db.get(key.encode("utf-8"))
+
+        if last_block:
+            last_block = last_block.decode("utf-8")
+
+        return last_block
+
+    def load_watcher_appointment(self, uuid):
+        """
+        Loads an appointment from the database using ``WATCHER_PREFIX`` as prefix to the given ``uuid``.
+
+        Args:
+            uuid (:obj:`str`): the appointment's unique identifier.
+
+        Returns:
+            :obj:`dict`: A dictionary containing the appointment data if they ``key`` is found.
+
+            Returns ``None`` otherwise.
+        """
+
+        try:
+            data = self.load_entry(uuid, prefix=WATCHER_PREFIX)
+            data = json.loads(data)
+        except (TypeError, json.decoder.JSONDecodeError):
+            data = None
+
+        return data
+
+    def load_responder_tracker(self, uuid):
+        """
+        Loads a tracker from the database using ``RESPONDER_PREFIX`` as a prefix to the given ``uuid``.
+
+        Args:
+            uuid (:obj:`str`): the tracker's unique identifier.
+
+        Returns:
+            :obj:`dict`: A dictionary containing the tracker data if they ``key`` is found.
+
+            Returns ``None`` otherwise.
+        """
+
+        try:
+            data = self.load_entry(uuid, prefix=RESPONDER_PREFIX)
+            data = json.loads(data)
+        except (TypeError, json.decoder.JSONDecodeError):
+            data = None
+
+        return data
+
+    def load_watcher_appointments(self, include_triggered=False):
+        """
+        Loads all the appointments from the database (all entries with the ``WATCHER_PREFIX`` prefix).
+
+        Args:
+            include_triggered (:obj:`bool`): whether to include the appointments flagged as triggered or not. ``False``
+                by default.
+
+        Returns:
+            :obj:`dict`: A dictionary with all the appointments stored in the database. An empty dictionary is there
```

Comment:
typo "if there are none"; or you could remove the last sentence - I find it redundant.